thetagang/portfolio_manager.py

```diff
@@ -290,103 +149,99 @@ def position_can_be_closed(self, position: PortfolioItem, table: Table) -> bool:
     def put_can_be_closed(self, put: PortfolioItem, table: Table) -> bool:
         return self.position_can_be_closed(put, table)
 
-    def put_can_be_rolled(self, put: PortfolioItem, table: Table) -> bool:
+    async def put_can_be_rolled(self, put: PortfolioItem, table: Table) -> bool:
         # Ignore long positions, we only roll shorts
         if put.position > 0:
             return False
 
         if not trading_is_allowed(self.config, put.contract.symbol):
             return False
 
-        if (
-            isinstance(put.contract, Option)
-            and self.put_is_itm(put.contract)
-            and self.config["roll_when"]["puts"]["always_when_itm"]
-        ):
-            table.add_row(
-                f"{put.contract.localSymbol}",
-                "[blue]Roll",
-                f"[blue]Will be rolled because put is ITM "
-                f"and always_when_itm={self.config['roll_when']['puts']['always_when_itm']}",
-            )
-            return True
-
-        # Check if this put is ITM, and if it's o.k. to roll
-        if (
-            not self.config["roll_when"]["puts"]["itm"]
-            and isinstance(put.contract, Option)
-            and self.put_is_itm(put.contract)
-        ):
-            return False
-
-        # Don't roll if there are excess puts and we're configured not to roll
-        if (
-            put.contract.symbol in self.has_excess_puts
-            and not self.config["roll_when"]["puts"]["has_excess"]
-        ):
-            table.add_row(
-                f"{put.contract.localSymbol}",
-                "[cyan1]None",
-                "[cyan1]Won't be rolled because there are excess puts",
-            )
-            return False
-
-        dte = option_dte(put.contract.lastTradeDateOrContractMonth)
-        pnl = position_pnl(put)
-
-        roll_when_dte = self.config["roll_when"]["dte"]
-        roll_when_pnl = self.config["roll_when"]["pnl"]
-        roll_when_min_pnl = self.config["roll_when"]["min_pnl"]
-
-        if (
-            "max_dte" in self.config["roll_when"]
-            and dte > self.config["roll_when"]["max_dte"]
-        ):
-            return False
+        try:
+            itm = await self.put_is_itm(put.contract)
```

Comment:
and move the `await self.put_is_itm(put.contract)` to one statement, so we don't need to evaluate this twice